lua/ttt2/libraries/bodysearch.lua

```diff
@@ -0,0 +1,1007 @@
+					params = {num = data.c4CutWire}
+				}}
+			}
+		end,
+		dmg = function(data)
+			if not data.dmgType then return end
+
+			local rawText = {
+				title = {
+					body = "search_title_dmg_" .. DamageToText(data.dmgType),
+					params = {amount = data.lastDamage}
+				},
+				text = {{
+					body = "search_dmg_" .. DamageToText(data.dmgType),
+					params = nil
+				}}
+			}
+
+			if data.killOrientation ~= CORPSE_KILL_NONE then
+				rawText.text[#rawText.text + 1] = {
+					body = orientationToText[data.killOrientation],
+					params = nil
+				}
+			end
+
+			if data.wasHeadshot then
+				rawText.text[#rawText.text + 1] = {
+					body = "search_head",
+					params = nil
+				}
+			end
+
+			return rawText
+		end,
+		wep = function(data)
+			if not data.wep then return end
+
+			local wep = util.WeaponForClass(data.wep)
+
+			local wname = wep and wep.PrintName
+
+			if not wname then return end
+
+			local rawText = {
+				title = {
+					body = wname,
+					params = nil
+				},
+				text = {{
+					body = "search_weapon",
+					params = {weapon = wname}
+				}}
+			}
+
+			if data.dist ~= CORPSE_KILL_NONE then
+				rawText.text[#rawText.text + 1] = {
+					body = distanceToText[data.killDistance],
+					params = nil
+				}
+			end
+
+			if data.killHitGroup > 0 then
+				rawText.text[#rawText.text + 1] = {
+					body = hitgroup_to_text[data.killHitGroup],
+					params = nil
+				}
+			end
+
+			return rawText
+		end,
+		death_time = function(data)
+			if not data.deathTime then return end
+
+			return {
+				title = {
+					body = "search_title_time",
+					params = nil
+				},
+				text = {{
+					body = "search_time",
+					params = nil
+				}}
+			}
+		end,
+		dna_time = function(data)
+			if not data.sampleDecayTime or data.sampleDecayTime - CurTime() <= 0 then return end
+
+			return {
+				title = {
+					body = "search_title_dna",
+					params = nil
+				},
+				text = {{
+					body = "search_dna",
+					params = nil
+				}}
+			}
+		end,
+		kill_list = function(data)
+			if not data.kills then return end
+
+			local num = table.Count(data.kills)
+
+			if num == 1 then
+				local vic = Entity(data.kills[1])
+				local dc = data.kills[1] == -1 -- disconnected
+
+				if dc or IsValid(vic) and vic:IsPlayer() then
+					return {
+						title = {
+							body = "search_title_kills",
+							params = nil
+						},
+						text = {{
+							body = "search_kills1",
+							params = {player = dc and "<Disconnected>" or vic:Nick()}
+						}}
+					}
+				end
+			elseif num > 1 then
+				local nicks = {}
+
+				for k, idx in pairs(data.kills) do
+					local vic = Entity(idx)
+					local dc = idx == -1
+
+					if dc or IsValid(vic) and vic:IsPlayer() then
+						nicks[#nicks + 1] = dc and "<Disconnected>" or vic:Nick()
+					end
+				end
+
+				return {
+					title = {
+						body = "search_title_kills",
+						params = nil
+					},
+					text = {{
+						body = "search_kills2",
+						params = {player = table.concat(nicks, "\n", 1, last)}
```

Comment:
`last` is an unset global.

Reply:
this was taken over from the old UI and after thinking about it I see that it makes no sense anyway. Even though it was defined in the original code